executor/include/execution/trait/can_require.hpp

```diff
@@ -14,17 +15,19 @@ template <typename Executor, typename Property,
                   Property::is_requirable &&
                   Property::template static_query<Executor>(),
               int> = 0>
-Executor require(Executor ex, const Property t) {
-  return ex.require(t);
+constexpr auto require(Executor &&ex, const Property &p) noexcept {
+  return Property::template static_query<Executor>();
```

Comment:
This looks weird. The original implementation made sense to me, now the function is not "requiring" anything. It is just making a query. Maybe rename to express better the intent, or  use your free-form query function.

Reply:
Thanks for the catch. I messed up the implementation when fixing `query()`. Will switch back to the original implementation.